### DISCUSSION TITLE
Rogue "about" directory and redirect loop

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [statiqdev/Discussions#92] [Q&A]
When building using `dotnet run -- publish`, a rogue "about" directory ends up in my output.

The input folder in my project has an `about.cshtml` but it does not have a folder of that name. However, when building, it appears (as an empty folder) in my output.

This is causing the site to return a 404 when published because the first htaccess rule it hits is one which looks for directories and appends `/index.html`.

Here's the relevant htaccess section:

```htaccess
# Navigate to html file
RewriteCond %{SCRIPT_FILENAME}.html -f
RewriteRule ^(.*)$ $1.html [L]


# Navigate to index file
RewriteCond %{SCRIPT_FILENAME} -d
RewriteRule ^(.*)$ $1/index.html [L]
```

Deleting the directory causes a 500 error with the following error message:
> Request exceeded the limit of 10 internal redirects due to probable configuration error.

I'm unsure where the recursion is happening. It should pass on the first htaccess condition but fail the second condition, just like accessing any other html file on the site (and I should note, all others work).

**Comment by daveaglick** [comment]
Let's focus on why you're getting that "about" directory since that's what's tripping up the `.htaccess` rules - regardless of how it's ending up in a recursive loop, it sounds like the part Statiq is responsible for is that empty directory.

My first instinct is that the empty folder is being generated to contain an `index.html` for extensionless routing - I.e. the convention where a file at `/foo/index.html` appears as `/foo` (which is what I think you're `.htaccess file is enabling?). The odd thing is that the actual `index.html` isn't being output like it should.

What does the `about.cshtml` file look like? More specifically, does it contain any metadata as front matter or a sidecar file? And is this in Statiq Web or Statiq Framework w/ custom pipelines?

**Reply by TheBrambleShark** [reply]
> 
> 
> What does the `about.cshtml` file look like? More specifically, does it contain any metadata as front matter or a sidecar file? And is this in Statiq Web or Statiq Framework w/ custom pipelines?


Here is the up to date version of [about.cshtml](https://github.com/LuzFaltex/luzfaltex.com/blob/main/input/about.cshtml). The only front matter it contains is the page title. This is a Statiq Web project with no custom pipelines. I add a custom shortcode but that isn't used on this page.

**Reply by daveaglick** [reply]
I'm trying to figure out how you're even able to run `dotnet run -- publish`...in general everything after the `--` gets passed to the application by the `dotnet` muxer, and Statiq treats the first non-argument (I.e. not a `-` or `--`) as the name of the pipeline(s) to run and there's no "publish" pipeline:

![image](https://user-images.githubusercontent.com/1020407/129410770-11d122bb-2de5-45d9-8ab5-d86715fd416b.png)

**Reply by TheBrambleShark** [reply]
I keep mixing that up it seems.

I meant to say `dotnet run -- deploy`

**Reply by daveaglick** [reply]
Gotcha - yeah, I ended up with "deploy" to try and avoid confusion with the built-in `dotnet publish` command. I'll take another look to try and reproduce.
